src/db.ts

```diff
@@ -60,27 +66,50 @@ export const client = <T extends { Bindings: ClientBindings }>(c: Context<T> | {
 	if (c.env.HYPERDRIVE?.connectionString) {
 		connectionString = c.env.HYPERDRIVE.connectionString;
 	}
+
+	// TODO: this is a bit hacky, but cloudflare workers does not seem to have a better way of interacting with state across the request.
+	// We keep the drizzle clients and pools in the context so that we can reuse them across the same request and so that they are cleanup by the garbage collector
+	let drizzleClients = c.__drizzleClients;
+	if (!drizzleClients) {
+		drizzleClients = new Map();
+		c.__drizzleClients = drizzleClients;
+	}
+	let pools = c.__pools;
+	if (!pools) {
+		pools = new Map();
+		c.__pools = pools;
+	}
+
```

Comment:
Hello @ivpusic @norbertdurcansk I think you guys have have more context here and might know of a better way, any suggestions?